libnavigation-base/src/main/java/com/mapbox/navigation/base/routerefresh/MapboxRouteOptionsProvider.kt

```diff
@@ -1,21 +1,36 @@
-package com.mapbox.navigation.core.directions.session
+package com.mapbox.navigation.base.routerefresh
 
 import android.location.Location
 import com.mapbox.api.directions.v5.models.RouteOptions
 import com.mapbox.geojson.Point
 import com.mapbox.navigation.base.trip.model.RouteProgress
-import com.mapbox.navigation.core.trip.session.TripSession
 
-internal object AdjustedRouteOptionsProvider {
+/**
+ * Default implementation of [RouteOptionsProvider]
+ */
+class MapboxRouteOptionsProvider : RouteOptionsProvider {
 
-    private const val DEFAULT_REROUTE_BEARING_TOLERANCE = 90.0
+    companion object {
```

Comment:
nah, `companion object` is much more obviously and doesn't create hide class

Reply:
> doesn't create hide class

It's the other way around. Moving the `const` out from the `companion object` into the file would save 1 unnecessary class creation (micro-optimization) as outer class (file) is always created. Not a big deal though because as above-mentioned is a micro-optimization.

> `companion object` is much more obviously

I'm curious, why do you think so? 🤔 `const` sounds pretty self-explanatory to me 😅

Reply:
Companion object: 
<img width="730" alt="Screen Shot 2020-06-22 at 16 04 41" src="https://user-images.githubusercontent.com/8738058/85290726-27950280-b4a2-11ea-80be-8d974bb67c55.png">
vs const out of the class 
<img width="693" alt="Screen Shot 2020-06-22 at 16 03 42" src="https://user-images.githubusercontent.com/8738058/85290785-37144b80-b4a2-11ea-8902-fb9510558c19.png">
Second variant create hidden class, first not. It's only point why I prefer use companion into the class, moreover, Google uses `companion` in samples as well

Reply:
There's one "extra" class created in both cases, the `Companion` and the `Kt` respectively. In any case, I've checked the bytecode and the top-level one generates less code 😅